test-runner/test_module_send_telemetry.py

```diff
@@ -40,3 +42,37 @@ def test_module_send_event_to_iothub():
     module_client.disconnect()
     log_message("disconnecting eventhub client")
     eventhub_client.disconnect()
+
+
+@pytest.mark.asyncio
+@pytest.mark.testgroup_edgehub_module_client
+@pytest.mark.testgroup_iothub_module_client
+@pytest.mark.callsSendEvent
+async def test_module_send_event_to_iothub_async():
```

Comment:
Not entirely clear on why this has to be it's own function/coroutine. Couldn't a singular asynchronous coroutine handle both sync and async? Especially since, at this level of abstraction in the application, it's entirely blind to the actual implementation of the SDK specific API?

Reply:
As far as having a single test for sync and async is concerned, that needs a change in framework that Bert wants to handle. I know the idea in general (is based on a choice passed at runtime) but I guess Bert has some additional stuff that he wants to go along side that. I can discuss more on this after I am back as I am not making drastic changes to tests now.

Before making change to that framework , currently we have 2 different connections.
`module_client = await connections.connect_test_module_client_async()`
`module_client = connections.connect_test_module_client()`
Following these 2 lines you will see that it actually has 2 different clients. And hence currently we require 2 tests.